src/DonationForms/resources/app/utilities/generateRequestErrors.ts

```diff
@@ -7,7 +7,20 @@ import {__} from '@wordpress/i18n';
 const generateRequestErrors = (values: Record<string, any>, errors: object[], setError: UseFormSetError<any>) => {
     Object.entries(errors).forEach(([field, value]) => {
         if (Object.keys(values).includes(field)) {
-            setError(field, {message: Array.isArray(value) ? value[0] : value});
+            const fieldElement: HTMLInputElement = document.querySelector('input[name="' + field + '"]');
+            const canFocus = fieldElement && fieldElement.type !== 'hidden';
+
+            setError(field, {message: Array.isArray(value) ? value[0] : value}, {shouldFocus: canFocus});
+
+            if (!canFocus) {
+                // In fields that aren't inputs by default or are hidden inputs, we need to use this workaround because the "shouldFocus" option will not work in these cases.
+                if (!fieldElement) {
+                    const fieldElementContainer = document.querySelector('.givewp-fields-' + String(field)); //E.g: <div class="givewp-fields-giftAid">content...</div>
+                    fieldElementContainer && fieldElementContainer.scrollIntoView({behavior: 'smooth'});
```

Comment:
We can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here to simplify the implementation.

```suggestion
                    fieldElementContainer?.scrollIntoView({behavior: 'smooth'});
```